### PR TITLE
Refs [TASK] [36669] Init base project

### PR DESCRIPTION
**Pull request by luongth-2520**
**Related ticket**
<a href="https://edu-redmine.sun-asterisk.vn/issues/36669">#Task 36669 Init base project</a>

**WHAT**
Init base project

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,84 @@
...
+# Version control
+vcs.xml
 
-# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
-hs_err_pid*
+# lint
+lint/intermediates/
+lint/generated/
+lint/outputs/
+lint/tmp/
```
Comment (khoaqh-2470) [inline .gitignore]: enter

File: app/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (khoaqh-2470) [inline app/.gitignore]: above

File: app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+dependencies {
+
+    implementation 'androidx.core:core-ktx:1.5.0'
+    implementation 'androidx.appcompat:appcompat:1.3.0'
+    implementation 'com.google.android.material:material:1.3.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
+    testImplementation 'junit:junit:4.+'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.2'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+}
```
Comment (khoaqh-2470) [inline app/build.gradle]: above. tất cả file đều cách xuống 1 dòng nha

File: app/src/main/java/com/sun/unsplash_02/base/BaseActivity.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    lateinit var progressDialog: ProgressDialog
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(getLayoutId())
+
+        progressDialog = ProgressDialog(this)
+        progressDialog.setMessage("Loading")
+        progressDialog.setCancelable(false)
+        progressDialog.isIndeterminate = true
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash_02/base/BaseActivity.kt]: hard code, thêm vào strings xml 
scope function

File: app/src/main/java/com/sun/unsplash_02/data/source/remote/fetchjson/ParseDataWithJson.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    @Throws(Exception::class)
+    fun getJsonFromUrl(urlString: String?): String? {
+        val url = URL(urlString)
+        val httpURLConnection = url.openConnection() as HttpURLConnection
+        httpURLConnection.connectTimeout = TIME_OUT
+        httpURLConnection.readTimeout = TIME_OUT
+        httpURLConnection.requestMethod = METHOD_GET
+        httpURLConnection.doOutput = true
+        httpURLConnection.connect()
+
```
Comment (trinq-2471) [inline app/src/main/java/com/sun/unsplash_02/data/source/remote/fetchjson/ParseDataWithJson.kt]: remove blank
```suggestion
```

File: app/src/main/java/com/sun/unsplash_02/base/BaseActivity.kt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(getLayoutId())
+
+        progressDialog = ProgressDialog(this)
+        progressDialog.apply {
+            setMessage(getString(R.string.msg_loading))
+            setCancelable(false)
+            isIndeterminate = true
+        }
+
```
Comment (trinq-2471) [inline app/src/main/java/com/sun/unsplash_02/base/BaseActivity.kt]: remove blank
```suggestion
```

File: .github/workflows/pull_request_title_conventions.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  pull_request:
+    types: [edited, opened, synchronize, reopened]
+
+jobs:
+  title-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: huytq-2422/action-pr-title@master
+        with:
+          regex: '^[ ]*(refs|Refs)[ ]*\[(TASK|BUG|FEATURE|TEST|SUPPORT)*\][ ]*\[[0-9]*\][ ]*' # Regex the title should match.
```
Comment (khoaqh-2470) [inline .github/workflows/pull_request_title_conventions.yml]: enter

File: .sun-ci.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+              - Author: $CI_COMMIT_AUTHOR \n
+              - Pull request: https://$CI_GIT_PROVIDER.com/$CI_REPOSITORY/$(cat .pull_end_point) [/info]\" \
+                \"https://api.chatwork.com/v2/rooms/$ROOM_ID/messages\""
+    only:
+      branches:
+        - develop
+        - main
+        - master
+      events:
+        - pull_request
```
Comment (khoaqh-2470) [inline .sun-ci.yml]: above

File: app/src/main/java/com/sun/unsplash_02/data/source/remote/fetchjson/ParseDataWithJson.kt [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    @Throws(Exception::class)
+    fun getJsonFromUrl(urlString: String?): String? {
+        val url = URL(urlString)
+        val httpURLConnection = url.openConnection() as HttpURLConnection
+        httpURLConnection.connectTimeout = TIME_OUT
+        httpURLConnection.readTimeout = TIME_OUT
+        httpURLConnection.requestMethod = METHOD_GET
+        httpURLConnection.doOutput = true
+        httpURLConnection.connect()
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash_02/data/source/remote/fetchjson/ParseDataWithJson.kt]: scope function

File: app/src/main/java/com/sun/unsplash_02/data/source/remote/fetchjson/ParseDataWithJson.kt [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+            }
+        } catch (e: JSONException) {
+            e.printStackTrace()
+        }
+        return null
+    }
+
+    companion object {
+        private const val TIME_OUT = 15000
+        private val METHOD_GET: String? = "GET"
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash_02/data/source/remote/fetchjson/ParseDataWithJson.kt]:
```suggestion
        private val METHOD_GET = "GET"
```

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+    <background android:drawable="@drawable/ic_launcher_background" />
+    <foreground android:drawable="@drawable/ic_launcher_foreground" />
+</adaptive-icon>
```
Comment (khoaqh-2470) [inline app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml]: enter

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+    <background android:drawable="@drawable/ic_launcher_background" />
+    <foreground android:drawable="@drawable/ic_launcher_foreground" />
+</adaptive-icon>
```
Comment (khoaqh-2470) [inline app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml]: enter

**Review by khoaqh-2470 — APPROVED** [review]

File: app/build.gradle [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    id 'kotlin-android'
+}
+
+android {
+    compileSdk 30
+    buildToolsVersion "30.0.3"
+
+    defaultConfig {
+        applicationId "com.sun.unsplash_02"
+        minSdk 21
```
Comment (tuanlvt-1117) [inline app/build.gradle]:
```suggestion
        minSdk 23
```

File: app/src/main/res/layout/activity_main.xml [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+    tools:context=".screen.main.MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Hello World!"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/activity_main.xml]: remove code test

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**